tests/components/google_pubsub/test_init.py

```diff
@@ -0,0 +1,262 @@
+"""The tests for the Google Pub/Sub component."""
+from collections import namedtuple
+from datetime import datetime
+
+import pytest
+
+import homeassistant.components.google_pubsub as google_pubsub
+from homeassistant.components.google_pubsub import DateTimeJSONEncoder as victim
+from homeassistant.const import EVENT_STATE_CHANGED
+from homeassistant.core import split_entity_id
+from homeassistant.setup import async_setup_component
+
+import tests.async_mock as mock
+
+GOOGLE_PUBSUB_PATH = "homeassistant.components.google_pubsub"
+
+
+async def test_datetime():
+    """Test datetime encoding."""
+    time = datetime(2019, 1, 13, 12, 30, 5)
+    assert victim().encode(time) == '"2019-01-13T12:30:05"'
+
+
+async def test_no_datetime():
+    """Test integer encoding."""
+    assert victim().encode(42) == "42"
+
+
+async def test_nested():
+    """Test dictionary encoding."""
+    assert victim().encode({"foo": "bar"}) == '{"foo": "bar"}'
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5745133)